unstructured/ingest/cli/cmds/sharepoint.py

```diff
@@ -25,6 +25,9 @@
 class SharepointCliConfig(BaseConfig, CliMixin):
     client_id: t.Optional[str] = None
     client_cred: t.Optional[str] = None
+    permissions_application_id: t.Optional[str] = None
+    permissions_client_cred: t.Optional[str] = None
+    permissions_tenant: t.Optional[str] = None
```

Comment:
none of these overlap with site/client_id/client_cred? or thinking it might make sense to allow users to have a distinct app with access for these things?

Reply:
None of these overlap unfortunately. Each of these are created separately, and replacing any of them with its document ingestion correspondent causes failure

Only possible reduction of args here is parsing the `--site` argument to obtain `--permissions-tenant`, however I don't recommend it. If we want to do that, we'll have to find examples for all possible `--site` URL formats, and write a parsing logic that encapsulates all cases. Only covering the URL format we have can fail often for other users; and finding other possible URL formats will probably require creating different sites from scratch and trying to ingest from each of them

Reply:
Only possible reduction of args here is parsing the `--site` argument to obtain `--permissions-tenant`, however I don't recommend it. If we want to do that, we'll have to find examples for all possible `--site` URL formats, and write a parsing logic that covers all cases. Only covering the URL format we have can fail often for other users; and finding other possible URL formats will probably require creating different sites from scratch and trying to ingest from each of them. 

Example: 

Our `--site` format is: https://xxxxx-admin.sharepoint.com/
Our `--permissions-tenant` format is: xxxxx.onmicrosoft.com

Others'  `--site` formats might be: 
- https://xxxxx.sharepoint.com/
- https://xxxxx-<something-else\>.sharepoint.com/
- (any other format that we are unaware)

Others' `--permissions-tenant` formats might be: 
- (any other format that we are unaware)

It seems to me that giving user the ability to mention their tenant URL would be much safer here than assuming a format on their site & tenant URLs, and parsing one to another